nomad/structs/structs.go

```diff
@@ -7255,6 +7255,10 @@ func (t *Task) Validate(ephemeralDisk *EphemeralDisk, jobType string, tgServices
 			mErr.Errors = append(mErr.Errors, fmt.Errorf("CSIPluginConfig PluginType must be one of 'node', 'controller', or 'monolith', got: \"%s\"", t.CSIPluginConfig.Type))
 		}
 
+		if t.CSIPluginConfig.HealthTimeout == 0 {
+			t.CSIPluginConfig.HealthTimeout = 30 * time.Second
```

Comment:
Added the default value as 30 seconds for backward compatibility. Let me know if this is the right spot for this @tgross

Reply:
There are two issues with this spot:

* This will help for ensuring backwards compatibility during upgrades, where the server might be newer than the client (although we don't recommend upgrading in that order). But we also need to account for cases where the job is already in the state store and lives across an upgrade. Those plugins won't hit `Validate()` again and so when the plugin probe reattaches we'll get 0.

* Also, although this technically works because we pass a pointer into the `Validate()` methods, we don't want to mutate the job during validation. If you look at `nomad/job_endpoint.go` you'll see a reference to both "admission mutators" and "admission validators". (This is mostly for architectural clarity in job registration.)

So let's move this down into the `csiPluginSupervisorHook` constructor. That'll solve both the problems above.

Reply:
Sounds good to me :+1: